File: problem_1.c
#include <stdio.h>

int main(void) 
{
    int i, j, k;

    for (i = 0; i < 10 ; i++) {
        for (j = 0; j < 10 ; j++) {
            for (k = 0; k < 10 ; k++) {
                if (((i != j) && (i != k) && (j != k)) && (3*(i*100 + j*10 + k) == j*111)) {
                    printf("Hundreds Position:          %d\n", i);
                    printf("Tens Position:              %d\n", j);
                    printf("Ones Position:              %d\n", k);
                    printf("Product of Three Positions: %d\n", i*j*k);
                }
            }
        }
    }
}

File: problem_1.py
for i in range(10):
    for j in range(10):
        for k in range(10):
            if (i != j) and (i != k) and (j != k) and (3*(i*100 + j*10 + k) == j*111): 
                print("Hundreds Position:           ", i)
                print("Tens Position:               ", j)
                print("Ones Position:               ", k)
                print("Product of Three Positions: ", i*j*k)